nutype_macros/src/string/parse.rs

```diff
@@ -11,80 +11,163 @@ use darling::util::SpannedValue;
+    let maybe_default_value = maybe_default_meta.map(|meta| quote::quote!(#meta));
+
     Ok(Attributes {
         new_unchecked,
         guard,
         maybe_default_value,
     })
 }
 
-fn parse_raw_attributes(input: TokenStream) -> Result<Attributes<StringRawGuard>, darling::Error> {
-    parse_nutype_attributes(parse_sanitize_attrs, parse_validate_attrs)(input)
-}
-
-fn parse_sanitize_attrs(
-    stream: TokenStream,
-) -> Result<Vec<SpannedStringSanitizer>, darling::Error> {
-    let attr_args = NestedMeta::parse_meta_list(stream)?;
+impl FromMeta for SpannedStringSanitizers {
+    fn from_list(items: &[NestedMeta]) -> darling::Result<Self> {
+        let mut errors = darling::Error::accumulator();
 
-    let mut errors = darling::Error::accumulator();
+        let raw_sanitizers: Vec<ParseableStringSanitizer> = items
+            .iter()
+            .flat_map(|arg| {
+                let res = ParseableStringSanitizer::from_list(std::slice::from_ref(arg));
+                errors.handle(res)
+            })
+            .collect();
 
-    let raw_sanitizers: Vec<ParseableStringSanitizer> = attr_args
-        .iter()
-        .flat_map(|arg| {
-            let res = ParseableStringSanitizer::from_list(std::slice::from_ref(arg));
-            errors.handle(res)
-        })
-        .collect();
+        let raw_sanitizers = errors.finish_with(raw_sanitizers)?;
 
-    let raw_sanitizers = errors.finish_with(raw_sanitizers)?;
+        let sanitizers: Vec<SpannedStringSanitizer> = raw_sanitizers
+            .into_iter()
+            .flat_map(ParseableStringSanitizer::into_spanned_string_sanitizer)
+            .collect();
 
-    let sanitizers: Vec<SpannedStringSanitizer> = raw_sanitizers
-        .into_iter()
-        .flat_map(ParseableStringSanitizer::into_spanned_string_sanitizer)
-        .collect();
+        Ok(Self(sanitizers))
+    }
 
-    Ok(sanitizers)
+    fn from_word() -> darling::Result<Self> {
+        // Provide a better error message than the default implementation
+        let msg = concat!(
+            "`sanitize` must be used with parenthesis.\n",
+            "For example:\n",
+            "\n",
+            "    sanitize(trim)\n\n"
+        );
+        let error = darling::Error::custom(msg);
+        Err(error)
+    }
 }
 
-fn parse_validate_attrs(
-    stream: TokenStream,
-) -> Result<Vec<SpannedStringValidator>, darling::Error> {
-    let attr_args = NestedMeta::parse_meta_list(stream)?;
+impl FromMeta for SpannedStringValidators {
+    fn from_list(items: &[NestedMeta]) -> darling::Result<Self> {
+        let mut errors = darling::Error::accumulator();
 
-    let mut errors = darling::Error::accumulator();
+        let parseable_validators: Vec<ParseableStringValidator> = items
+            .iter()
+            .flat_map(|arg| {
+                let res = ParseableStringValidator::from_list(std::slice::from_ref(arg));
+                errors.handle(res)
+            })
+            .collect();
```

Comment:
Why do you collect only to call `into_iter()`?

Reply:
Good point.
The reason is that Rust  would not allow 2 mutable borrows of  `errors` .